crates/sui-indexer-alt/src/handlers/sum_coin_balances.rs

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    collections::{btree_map::Entry, BTreeMap},
+    sync::Arc,
+};
+
+use anyhow::{anyhow, bail, ensure};
+use diesel::{upsert::excluded, ExpressionMethods};
+use diesel_async::RunQueryDsl;
+use futures::future::try_join_all;
+use sui_types::{
+    base_types::ObjectID, effects::TransactionEffectsAPI, full_checkpoint_content::CheckpointData,
+    object::Owner,
+};
+
+use crate::{
+    db,
+    models::objects::{StoredObjectUpdate, StoredSumCoinBalance},
+    pipeline::{sequential::Handler, Processor},
+    schema::sum_coin_balances,
+};
+
+/// Each insert or update will include at most this many rows -- the size is chosen to maximize the
+/// rows without hitting the limit on bind parameters.
+const UPDATE_CHUNK_ROWS: usize = i16::MAX as usize / 5;
```

Comment:
ditto, good to read `5` from schema

Reply:
We can go back over and provide some automation to calculate the number of fields based on model types, using a procedural macro:

https://linear.app/mysten-labs/issue/DVX-447/automatically-calculate-field-count